spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigUtils.java

```diff
@@ -251,12 +269,156 @@ public static boolean noSources(List<?> sources, String namespace) {
 		return false;
 	}
 
+	public static LinkedHashSet<StrictProfile> profiles(boolean includeProfileSpecificSources,
```

Comment:
this is one of the core changes. I get all active profiles and profiles that were specified under `strict-for-profiles`. The first two if statements should be obvious, if you said : `include-profile-specific-sources=false`, then you get nothing back.

- if there is nothing under `strict-for-profiles`, then everything is non-strict.
- if you said `strcit-for-profiles=dev`, but `dev` is not active profile - we fail.
- if there are 4 profiles : `a`, `b`, `c`, `d`. and `strcit-for-profiles: d, c`; the result of this method is (logically) : `a`, `c`, `d`, `c`. i.e. `strict-for-profiles` order is enforced and is dealt with _after_ any other (optional) source/profile.